deepcell/image_generators/semantic.py

```diff
@@ -1,387 +1,403 @@
+#
+#     http://www.github.com/vanvalenlab/deepcell-tf/LICENSE
+#
+# The Work provided may be used for non-commercial academic purposes only.
+# For any other use of the Work, including commercial use, please contact:
+# vanvalenlab@gmail.com
+#
+# Neither the name of Caltech nor the names of its contributors may be used
+# to endorse or promote products derived from this software without specific
+# prior written permission.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Semantic segmentation data generators."""
+
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import division
+
+import os
+
+import numpy as np
+
+from tensorflow.python.keras import backend as K
+from tensorflow.python.keras.preprocessing.image import array_to_img
+from tensorflow.python.keras.preprocessing.image import Iterator
+from tensorflow.python.keras.preprocessing.image import ImageDataGenerator
+from tensorflow.python.platform import tf_logging as logging
+
+try:
+    import scipy
+    # scipy.linalg cannot be accessed until explicitly imported
+    from scipy import linalg
+    # scipy.ndimage cannot be accessed until explicitly imported
+    from scipy import ndimage
+except ImportError:
+    scipy = None
+
+from deepcell.image_generators import _transform_masks
+
+
+class SemanticIterator(Iterator):
+    """Iterator yielding data from Numpy arrays (X and y).
+
+    Args:
+        train_dict (dict): Consists of numpy arrays for X and y.
+        image_data_generator (ImageDataGenerator): For random transformations
+            and normalization.
+        batch_size (int): Size of a batch.
+        min_objects (int): Images with fewer than 'min_objects' are ignored.
+        shuffle (bool): Whether to shuffle the data between epochs.
+        seed (int): Random seed for data shuffling.
+        data_format (str): One of 'channels_first', 'channels_last'.
+        save_to_dir (str): Optional directory where to save the pictures
+            being yielded, in a viewable format. This is useful
+            for visualizing the random transformations being
+            applied, for debugging purposes.
+        save_prefix (str): Prefix to use for saving sample
+            images (if save_to_dir is set).
+        save_format (str): Format to use for saving sample images
+            (if save_to_dir is set).
+    """
+    def __init__(self,
+                 train_dict,
+                 image_data_generator,
+                 batch_size=1,
+                 shuffle=False,
+                 transforms=['watershed-cont'],
+                 transforms_y2=['pixelwise'],
```

Comment:
For some reason git is freaking out and said the entire file changed. First change is here